evm/src/arithmetic/add.rs

```diff
@@ -47,7 +47,7 @@ where
     for (a, b) in larger.zip(smaller) {
         // t should be either 0 or 2^LIMB_BITS
         let t = cy + a - b;
-        yield_constr.constraint(is_op * t * (overflow - t));
+        yield_constr.constraint_transition(is_op * t * (overflow - t));
```

Comment:
The issue with this is that the constraint will be disabled on the very last row. So it works on modulo operations (because `lv` will never be the last row) but it's not sound for e.g. plain addition or multiplication (because they only take up one row, which is permitted to be the last).

Possible solutions:
- Have two versions of the function
- Let the user choose between `constraint` and `constraint_transition` through one of:
  - Accept a closure as an argument
  - Put a flag in the arguments
  - Flag again, but as a `const` generic

Reply:
These are all somewhat ugly; I chose to pass a flag in the arguments as it seems the simplest.